openwebstart/src/main/java/com/openwebstart/jvm/RemoteRuntimeManager.java

```diff
@@ -35,23 +36,23 @@
     private RemoteRuntimeManager() {
     }
 
-    public Optional<RemoteJavaRuntime> getBestRuntime(final VersionString versionString, final URI specificServerEndpoint, final String vendor) throws Exception {
+    public Optional<RemoteJavaRuntime> getBestRuntime(final VersionString versionString, final URL specificServerEndpoint, final String vendor) throws Exception {
         return getBestRuntime(versionString, specificServerEndpoint, vendor, OperationSystem.getLocalSystem());
     }
 
-    public Optional<RemoteJavaRuntime> getBestRuntime(final VersionString versionString, final URI specificServerEndpoint, final String vendor, final OperationSystem operationSystem) throws Exception {
+    public Optional<RemoteJavaRuntime> getBestRuntime(final VersionString versionString, final URL specificServerEndpoint, final String vendor, final OperationSystem operationSystem) throws Exception {
         Assert.requireNonNull(versionString, "versionString");
-        Assert.requireNonBlank(vendor, "vendor");
         Assert.requireNonNull(operationSystem, "operationSystem");
 
-        LOG.debug("Trying to find remote Java runtime. Requested version: '" + versionString + "' Requested vendor: '" + vendor + "' requested os: '" + operationSystem + "'");
+        final String vendorName = RuntimeManagerConfig.isNonDefaultVendorsAllowed() && !isBlank(vendor) ? vendor : RuntimeManagerConfig.getDefaultVendor();
+        final Vendor vendorForRequest = Vendor.fromString(vendorName);
 
-        final URI endpointForRequest = Optional.ofNullable(specificServerEndpoint)
+        LOG.debug("Trying to find remote Java runtime. Requested version: '" + versionString + "' Requested vendor: '" + vendorForRequest + "' requested os: '" + operationSystem + "'");
```

Comment:
consider to use {} instead of string concatenation for consistency to other log statements in this PR and better readability here